make/test.mk

```diff
@@ -199,3 +210,23 @@ else
 endif
 	sed -e 's|REPLACE_IMAGE|${IMAGE_NAME}|g' ${E2E_REPO_PATH}/deploy/operator.yaml | oc apply -f -
 
+.PHONY: build-and-deploy-deployment
+build-and-deploy-deployment:
+# when e2e tests are triggered from different repo - eg. as part of PR in host-operator repo - and the image of the deployment is (not) provided
+ifeq ($(SET_IMAGE_NAME),)
+	# check if it is running locally
+    ifeq ($(OPENSHIFT_BUILD_NAMESPACE),)
+		# if it is running locally, then build the image via docker
+		$(eval IMAGE_NAME := docker.io/${GO_PACKAGE_ORG_NAME}/${REPO_NAME}:${GIT_COMMIT_ID_SHORT})
```

Comment:
We will need to stop using docker.io and switch to quay.io instead. We use docker.io currently only for local images (we don't really push anything to docker.io) but still would be better to switch to quay.io.
But it's OK do it in a separate PR.

Reply:
it's partially already done - we check the OpenShift version (if it is 3 or 4) and based on that we use either quay.io (for OS 4) or docker.io (for OS 3 - minishift) for the local testing. https://github.com/codeready-toolchain/toolchain-e2e/blob/master/make/test.mk#L190-L204 
But I'm fine with dropping docker.io completely (we will need to do it anyway as soon as minishift is not usable/compatible enough)

Reply:
even for minishift we don't need docker.io. Currently we use docker.io only for local images (for running them in minishift). The images which are never pushed to docker.io. So, it doesn't matter if we use docker.io or quay.io or whatever in the image names for local development :)

Reply:
using `quay.io` at least for reg_service deployment which is what we are introducing with this PR.  See commit https://github.com/codeready-toolchain/toolchain-e2e/pull/12/commits/027176b4ceb7eecfd996451ff4cf5a92c72db3bb